.github/workflows/check-odo.yml

```diff
@@ -4,42 +4,122 @@ on:
-      - name: Get latest ODO version
-        run: |
-          echo "REPO_ODO_VERSION=$(cat src/tools.json | jq -r .odo.version)" >> $GITHUB_ENV
-          LATEST_TOOL_RELEASE_RESP=$(gh release --repo ${{ env.TOOL_REPO }} view --json tagName,url)
-          echo "LATEST_TOOL_RELEASE=$(echo ${LATEST_TOOL_RELEASE_RESP} | jq -r .tagName | sed 's|v||')" >> $GITHUB_ENV
-          echo "LATEST_TOOL_URL=$(echo ${LATEST_TOOL_RELEASE_RESP} | jq -r .url)" >> $GITHUB_ENV
-      - name: Find existing PR for ODO version
-        run: |
-          echo PR_EXISTS=$(gh pr --repo ${{ github.repository }} list --state all --search "update odo ${{env.LATEST_TOOL_RELEASE}} in:title" --json url | jq length) >> $GITHUB_ENV
       - name: Update src/tools.json with latest odo version
-        if: ${{ (env.LATEST_TOOL_RELEASE != '') && (env.LATEST_TOOL_RELEASE != env.REPO_ODO_VERSION) && (env.PR_EXISTS == 0) }}
         run: |
-          jq --indent 4 '.odo.version = "${{ env.LATEST_TOOL_RELEASE }}"' src/tools.json | jq --indent 4 '.odo.versionRange = "^${{ env.LATEST_TOOL_RELEASE }}"' | jq --indent 4 '.odo.versionRangeLabel = "version >= ${{ env.LATEST_TOOL_RELEASE }}"' > src/tools.json.new
+          jq --indent 4 '.odo.description = "ODO CLI tool"' src/tools.json \
+            | jq --indent 4 '.odo.vendor = "Red Hat Developer"' \
+            | jq --indent 4 '.odo.name = "odo"' \
+            | jq --indent 4 '.odo.version = "${{ needs.check-odo.outputs.last-release-version }}"' \
+            | jq --indent 4 '.odo.versionRange = "^${{ needs.check-odo.outputs.last-release-version }}"' \
+            | jq --indent 4 '.odo.versionRangeLabel = "version >= ${{ needs.check-odo.outputs.last-release-version }}"' > src/tools.json.new
           mv src/tools.json.new src/tools.json
           for platform in win32 darwin darwin-arm64 linux linux-arm64; do
-            old_url=`jq -r .odo.platform.${platform}.url src/tools.json`
-            new_url=`echo ${old_url} | sed "s|${{ env.REPO_ODO_VERSION }}|${{ env.LATEST_TOOL_RELEASE }}|"`
+            pltfrm="$platform"
+            ext=".tar.gz"
+            exeExt=""
+            if [[ "$platform" == "win"* ]]; then
+              pltfrm="windows"
+              ext=".zip"
+              exeExt=".exe"
+            fi
+            arch="-amd64"
+            if [[ $platform == *"-a"* ]]; then
+              arch="" # already in platform string
+            fi
+            old_url=`jq -r ".odo.platform[\"${platform}\"].url" src/tools.json`
+            new_url="https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-nightly-builds/odo-${pltfrm}${arch}-${{ needs.check-odo.outputs.last-commit-id }}${ext}"
             checksum=`curl -s ${new_url}.sha256`
-            jq --indent 4 ".odo.platform.${platform}.url = \"${new_url}\"" src/tools.json | jq --indent 4 ".odo.platform.${platform}.sha256sum = \"${checksum}\"" > src/tools.json.new
+            dlFileName="odo-${pltfrm}${arch}${ext}"
+            cmdFileName="odo${exeExt}"
```

Comment:
https://github.com/redhat-developer/vscode-openshift-tools/blob/eb674658b2c69e8977b454a30393a9fb9efe9cbb/src/downloadUtil/downloadBinaries.ts#L40

https://github.com/redhat-developer/vscode-openshift-tools/blob/eb674658b2c69e8977b454a30393a9fb9efe9cbb/src/downloadUtil/archive.ts#L14-L24

https://github.com/redhat-developer/vscode-openshift-tools/blob/eb674658b2c69e8977b454a30393a9fb9efe9cbb/src/downloadUtil/archive.ts#L75-L81

If you look at how `cmdFileName` is used, you'll see it is the name of the file within the archive to be extracted. However, if you check the URLs for each platform you'll see that the command file name embeds the abbreviated commit sha into the file name, so this needs to be adjusted for all platforms.

Reply:
`cmdFilename` will always be either `odo` (for Linux and Mac) or `odo.exe` for windows. 
So, I believe it's adjusted